.drone.yml

```
@@ -3,168 +3,523 @@ kind: pipeline
+    pull: if-not-exists
+    image: 340268328991.dkr.ecr.eu-west-2.amazonaws.com/acp/dind
+    environment:
+      DOCKER_PASSWORD:
+        from_secret: DOCKER_PASSWORD
+    commands:
+      - docker login -u="ukhomeofficedigital+modern_slavery_bot" -p=$${DOCKER_PASSWORD} quay.io
+      - docker tag $${IMAGE_REPO}:$${DRONE_COMMIT_SHA} $${IMAGE_URL}/$${IMAGE_REPO}:$${DRONE_COMMIT_SHA}
+      - docker push $${IMAGE_URL}/$${IMAGE_REPO}:$${DRONE_COMMIT_SHA}
+    when:
+      branch: master
+      event: [push, pull_request]
+
+  # Deploy to a branch via a pull request
+  - name: deploy_to_branch
+    pull: if-not-exists
+    image: quay.io/ukhomeofficedigital/kd:v1.14.0
+    environment:
+      KUBE_SERVER_DEV:
+        from_secret: KUBE_SERVER_DEV
+      KUBE_TOKEN_DEV:
+        from_secret: DRONE_KUBE_TOKEN
+    commands:
+      - export KUBE_TOKEN=$${DRONE_KUBE_TOKEN}
+      - export KUBE_NAMESPACE=$${BRANCH_ENV}
+      - export KUBE_SERVER_DEV=$${KUBE_SERVER_DEV}
+      - export KUBE_TOKEN_DEV=$${DRONE_KUBE_TOKEN}
+      - export INGRESS_INTERNAL_ANNOTATIONS=$${HOF_CONFIG}/ingress-internal-annotations.yaml
+      - export CONFIGMAP_VALUES=$${HOF_CONFIG}/configmap-values.yaml
+      - export NGINX_SETTINGS=$${HOF_CONFIG}/nginx-settings.yaml
+      - export DRONE_SOURCE_BRANCH=$(echo $${DRONE_SOURCE_BRANCH} | tr '[:upper:]' '[:lower:]')
+      - kd --insecure-skip-tls-verify --check-interval 10s -f kube/redis-service.yaml -f kube/redis-network-policy.yaml -f kube/redis-deployment.yaml
+      - kd --insecure-skip-tls-verify -f kube/configmap.yaml -f kube/ingress-internal.yaml -f kube/networkpolicy-internal.yaml -f kube/service.yaml
+      - kd --insecure-skip-tls-verify --timeout 10m --check-interval 10s -f kube/deployment.yaml
+      - sleep $${READY_FOR_TEST_DELAY}
+      - echo "UAT Branch - $${APP_NAME}-$${DRONE_SOURCE_BRANCH}.branch.internal.ms-notprod.homeoffice.gov.uk"
+    when:
+      branch: master
+      event: pull_request
+  
+  # - name: linting_branch
+  #   <<: *linting
+  #   when:
+  #     branch:
+  #       include:
+  #         - master
+  #         - feature/*
+  #     event: pull_request
+  
+  # Deploy to Master UAT environment
+  - name: deploy_to_dev
+    pull: if-not-exists
+    image: quay.io/ukhomeofficedigital/kd:v1.14.0
+    environment:
+      KUBE_SERVER:
+        from_secret: KUBE_SERVER_DEV
+      DRONE_NOTPROD_TOKEN:
+        from_secret: DRONE_NOTPROD_TOKEN
+    commands:
+      - export KUBE_TOKEN=$${DRONE_NOTPROD_TOKEN}
+      - export KUBE_TOKEN_DEV=$${DRONE_NOTPROD_TOKEN}
+      - export KUBE_NAMESPACE=$${UAT_ENV}
+      - export INGRESS_INTERNAL_ANNOTATIONS=$${HOF_CONFIG}/ingress-internal-annotations.yaml
+      - export CONFIGMAP_VALUES=$${HOF_CONFIG}/configmap-values.yaml
+      - export NGINX_SETTINGS=$${HOF_CONFIG}/nginx-settings.yaml
+      - kd --insecure-skip-tls-verify --check-interval 10s -f kube/redis-service.yaml -f kube/redis-network-policy.yaml -f kube/redis-deployment.yaml
```

You'd need to tweak this slightly but if you wanted to remove the verify skip you can pass the URL of the certificate:
```suggestion
      - export KUBE_CERTIFICATE_AUTHORITY="https://raw.githubusercontent.com/UKHomeOffice/acp-ca/master/acp-notprod.crt"
      - kd --check-interval 10s -f kube/redis-service.yaml -f kube/redis-network-policy.yaml -f kube/redis-deployment.yaml
```
```